src/protocolsupport/protocol/typeremapper/itemstack/toclient/ItemIdToPEIdSpecificRemapper.java

```diff
@@ -3,13 +3,32 @@
 import protocolsupport.api.ProtocolVersion;
 import protocolsupport.protocol.typeremapper.itemstack.ItemStackSpecificRemapper;
 import protocolsupport.protocol.typeremapper.pe.PEDataValues;
+import protocolsupport.utils.IntTuple;
 import protocolsupport.zplatform.itemstack.ItemStackWrapper;
 
 public class ItemIdToPEIdSpecificRemapper implements ItemStackSpecificRemapper {
 
 	@Override
 	public ItemStackWrapper remap(ProtocolVersion version, String locale, ItemStackWrapper itemstack) {
-		itemstack.setTypeId(PEDataValues.ITEM_ID.getRemap(itemstack.getTypeId()));
+		IntTuple itemAndData = PEDataValues.ITEM_ID.getRemap(itemstack.getTypeId(), itemstack.getData());
+
+		if (itemAndData == null) { // If itemAndData is null, then we are going to get the "-1" remapper
+			// When a remapper uses -1, it means it is a "remap every sub ID to this new ID"
```

Comment:
No need for comments like those here. Also with 1.13 this map will be gone, so no point to commenting the Complex remapper itself too